### PR TITLE
feat(swb-ref): add dynamicRouteConfig and update projectEnvService on…

### PR DESCRIPTION
**Pull request by kpark277**
… how IdentityPermission created

Issue #, if available:

Description of changes:

Checklist:

* [ ] Have you successfully deployed to an AWS account with your changes?
* [x] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: solutions/swb-reference/src/dynamicRouteConfig.ts [hunk 1]
```diff
@@ -0,0 +1,533 @@
...
+      }
+    ]
+  },
+  '/datasets': {
+    GET: [
+      {
+        action: 'READ',
+        subject: {
+          subjectType: SwbAuthZSubject.SWB_DATASET,
+          subjectId: '*'
```
Comment (htztomic) [inline solutions/swb-reference/src/dynamicRouteConfig.ts:118]: should this be limited in scope? This allowing reading of all datasets

File: solutions/swb-reference/src/dynamicRouteConfig.ts [hunk 2]
```diff
@@ -0,0 +1,533 @@
...
+        }
+      }
+    ],
+    POST: [
+      {
+        action: 'CREATE',
+        subject: {
+          subjectType: SwbAuthZSubject.SWB_DATASET,
+          subjectId: '*'
```
Comment (htztomic) [inline solutions/swb-reference/src/dynamicRouteConfig.ts:127]: same comment above

File: solutions/swb-reference/src/dynamicRouteConfig.ts [hunk 3]
```diff
@@ -0,0 +1,533 @@
...
+      }
+    ]
+  },
+  '/environments': {
+    GET: [
+      {
+        action: 'READ',
+        subject: {
+          subjectType: SwbAuthZSubject.SWB_ENVIRONMENT,
+          subjectId: '*'
```
Comment (htztomic) [inline solutions/swb-reference/src/dynamicRouteConfig.ts:173]: should this be limited in scope?

File: solutions/swb-reference/src/dynamicRouteConfig.ts (hunk 1 above)
Comment (kpark277) [reply solutions/swb-reference/src/dynamicRouteConfig.ts:118]: These will need to get refactored when @aws-tyler updates datasets.

File: solutions/swb-reference/src/constants.ts [hunk 4]
```diff
@@ -207,13 +207,18 @@ const endPointPrefix: string = 'ENDPOINT';
...
+  SWB_COST_CENTER = 'SWB_COST_CENTER',
   SWB_DATASET = 'SWB_DATASET',
   SWB_DATASET_ACCESS_LEVEL = 'SWB_DATASET_ACCESS_LEVEL',
   SWB_ENVIRONMENT = 'SWB_ENVIRONMENT',
   SWB_ENVIRONMENT_CONNECTION = 'SWB_ENVIRONMENT_CONNECTION',
   SWB_ENVIRONMENT_TYPE = 'SWB_ENVIRONMENT_TYPE',
   SWB_ETC = 'SWB_ETC',
   SWB_PROJECT = 'SWB_PROJECT',
+  SWB_PROJECT_USER_ASSOCIATION = 'SWB_PROJECT_USER_ASSOCIATION',
+  SWB_ROLE = 'SWB_ROLE',
```
Comment (htztomic) [inline solutions/swb-reference/src/constants.ts:221]: this isn't being used

Comment (kpark277) [reply solutions/swb-reference/src/constants.ts:221]: SWB_PROJECT_USER_ASSOCIATION is used by `/projects/:projectId/users/:userId`
Correct that SWB_ROLE isn't used, that will be deleted.

**Review by aws-tyler — APPROVED** [review]

**Pull request merged**